eng/common/templates/steps/source-build.yml

```diff
@@ -63,11 +63,6 @@ steps:
       targetRidArgs='/p:TargetRid=${{ parameters.platform.targetRID }}'
     fi
 
-    runtimeOsArgs=
-    if [ '${{ parameters.platform.runtimeOS }}' != '' ]; then
-      runtimeOsArgs='/p:RuntimeOS=${{ parameters.platform.runtimeOS }}'
-    fi
-
```

Comment:
@carlossanlop this is likely the change that causes the source build leg to fail

Reply:
So are you suggesting we revert this code removal?

Reply:
Responded offline. I'm just saying that this PR could be related to the Banana Source Build break, and you want to hold off merging it until you understand the break. A binlog should tell.

Reply:
No, this can't be reverted. It will get moved in a subsequent PR. This is a common script file.

Reply:
This change: https://github.com/dotnet/runtime/commit/552fcf0fcc47729e5d319a1c70182993a98af640 is the issue.

Basically, this change to the template must go into arcade.

/cc @ayakael

Reply:
I think for now a PR can be opened in arcade, and this change reverted in this PR.

Reply:
I reverted the source-build.yml change in this PR to unblock it.

@ayakael please submit your template change to the arcade repo directly.

Reply:
Copy, it's here: https://github.com/dotnet/arcade/pull/11604

Reply:
Ugh, the maestro bot brought these removals back, causing the banana RID failure happened in the last run.
I'll merge the PR with that known failure, and will submit a separate PR to get the lines restored. I don't want to restart the CI just for this.